### PR TITLE
Eradicating all existance of NO_DEDICATED macro

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Removes all traces of existence of the NO_DEDICATED macro

### CODE REVIEW

File: addons/sys_attenuate/XEH_preInit.sqf [hunk 1]
```diff
@@ -10,4 +8,6 @@ PREP_RECOMPILE_END;
 
 GVAR(attenuationCache) = HASH_CREATE;
 
-ADDON = true;
+if (!hasInterface) exitWith {
```
Comment (jonpas) [inline addons/sys_attenuate/XEH_preInit.sqf]: One thing I meant when I said streamline preInit files is things like this. In this case it should look like this for easiest readability (assuming I got the logic right):
```
#include "script_component.hpp"

ADDON = false;

PREP_RECOMPILE_START;
#include "XEH_PREP.hpp"
PREP_RECOMPILE_END;

if (hasInterface) then {
    GVAR(attenuationCache) = HASH_CREATE;
};

ADDON = true;
```

File: addons/sys_components/XEH_postInit.sqf [hunk 2]
```diff
@@ -1,2 +1,3 @@
 #include "script_component.hpp"
-NO_DEDICATED;
+
+if (!hasInterface) exitWith {};
```
Comment (jonpas) [inline addons/sys_components/XEH_postInit.sqf]: Hmm, we should just remove that file, it has no content. 😆

**Review by jonpas — CHANGES_REQUESTED** [review]
Partial review.

File: addons/sys_components/XEH_postInit.sqf (hunk 2 above)
Comment (TheMagnetar) [reply addons/sys_components/XEH_postInit.sqf]: Agreed, none of the other branches use this file in any anyway

File: addons/sys_attenuate/XEH_preInit.sqf (hunk 1 above)
Comment (TheMagnetar) [reply addons/sys_attenuate/XEH_preInit.sqf]: OK, This is the other way around. While this improves readability in shorter files, longer ones like sys_core (XEH_postInit), the solution taken here and started by snippers is more clear in my opinion.

**Comment by TheMagnetar** [comment]
This has been tested in a dedicated without RPT errors.

**Comment by jonpas** [comment]
Will merge for 2.5.0.

**Review by jonpas — APPROVED** [review]

File: addons/sys_antenna/XEH_postInit.sqf [hunk 3]
```diff
@@ -1,17 +1,18 @@
 #include "script_component.hpp"
-NO_DEDICATED;
 
-private _components = configFile >> "CfgAcreComponents";
+if (hasInterface) then {
```
Comment (jonpas) [inline addons/sys_antenna/XEH_postInit.sqf]: `if (!hasInterface) exitWith {};` would make more sense here, less indentation needed. It's only for preInit that we can't just exit early due to `˙ADDON = false/true;`.

File: addons/sys_attenuate/XEH_preInit.sqf [hunk 4]
```diff
@@ -1,7 +1,5 @@
 #include "script_component.hpp"
 
-NO_DEDICATED;
-
 ADDON = false;
 
 PREP_RECOMPILE_START;
```
Comment (jonpas) [inline addons/sys_attenuate/XEH_preInit.sqf:5]: `GVAR(attenuationCache) = HASH_CREATE;` should be inside `if (hasInterface) then {`

File: addons/sys_io/XEH_preInit.sqf [hunk 5]
```diff
@@ -9,11 +8,14 @@ PREP_RECOMPILE_START;
...
 PREP_RECOMPILE_END;
 
-DGVAR(pipeCode) = "0";
-DGVAR(ioEventFnc) = {};
-DGVAR(runserver) = false;
-DGVAR(serverStarted) = false;
-DGVAR(pongTime) = diag_tickTime;
-DGVAR(connectCount) = 15;
+if (hasInterface) then {
+
```
Comment (jonpas) [inline addons/sys_io/XEH_preInit.sqf]: No empty line.

File: addons/sys_list/XEH_postInit.sqf [hunk 6]
```diff
@@ -1,6 +1,8 @@
 #include "script_component.hpp"
-NO_DEDICATED;
-LOG("ADDING KEY HANDLERS FOR LIST");
 
-["ACRE2", "CycleRadio", (localize LSTRING(CycleRadio)), { [1] call FUNC(cycleRadios) }, "", [58, [true, false, true]]] call cba_fnc_addKeybind;
-["ACRE2", "OpenRadio", (localize LSTRING(OpenRadio)), { [ACRE_ACTIVE_RADIO] call EFUNC(sys_radio,openRadio); false }, "", [58, [false, true, true]]] call cba_fnc_addKeybind;
+if (hasInterface) then {
```
Comment (jonpas) [inline addons/sys_list/XEH_postInit.sqf]: Let's keep `!hasInterface -> exitWith` here as well.

File: addons/sys_prc343/XEH_preInit.sqf [hunk 7]
```diff
@@ -8,7 +8,9 @@ PREP_RECOMPILE_END;
 
 [] call FUNC(preset_information);
 
-NO_DEDICATED;
+if (!hasInterface) exitWith {
```
Comment (jonpas) [inline addons/sys_prc343/XEH_preInit.sqf]: `if (hasInterface)` around `GVAR` and `DGVAR`.

File: addons/sys_radio/XEH_postInit.sqf [hunk 8]
```diff
@@ -1,12 +1,9 @@
 #include "script_component.hpp"
 
-NO_DEDICATED;
+if (hasInterface) then {
```
Comment (jonpas) [inline addons/sys_radio/XEH_postInit.sqf]: Let's keep `!hasInterface -> exitWith` here as well.

File: addons/sys_radio/XEH_preInit.sqf [hunk 9]
```diff
@@ -10,7 +10,9 @@ PREP_RECOMPILE_END;
 GVAR(radioUniqueCache) = HASH_CREATE;
 GVAR(radioBaseClassCache) = HASH_CREATE;
 
-NO_DEDICATED;
+if (!hasInterface) exitWith {
```
Comment (jonpas) [inline addons/sys_radio/XEH_preInit.sqf]: Same as above, whatever is below here inside `if (hasInterface)`.

File: addons/sys_sem52sl/XEH_preInit.sqf [hunk 10]
```diff
@@ -15,7 +15,9 @@ GVAR(channelModulation) = "FM";
 GVAR(channelEncryption) = 0;
 GVAR(channelSquelch) = 0;
 
-NO_DEDICATED;
+if (!hasInterface) exitWith {
```
Comment (jonpas) [inline addons/sys_sem52sl/XEH_preInit.sqf]: Same as above.

File: addons/sys_server/XEH_postInitClient.sqf [hunk 11]
```diff
@@ -1,8 +1,8 @@
 #include "script_component.hpp"
 
-NO_DEDICATED;
+if (hasInterface) then {
```
Comment (jonpas) [inline addons/sys_server/XEH_postInitClient.sqf]: Let's keep `!hasInterface -> exitWith` here as well.

File: addons/sys_signal/XEH_preInit.sqf [hunk 12]
```diff
@@ -1,24 +1,24 @@
...
+    DGVAR(omnidirectionalRadios) = 0;
 
-FUNC(handleEndTransmission) = {
-    _data = _this select 2;
-    _transmitterClass = _data select 0;
-    missionNamespace setVariable [_transmitterClass + "_running_count", 0];
-    missionNamespace setVariable [_transmitterClass + "_best_signal", -992];
-    missionNamespace setVariable [_transmitterClass + "_best_ant", ""];
-    nil;
+    FUNC(handleEndTransmission) = {
```
Comment (jonpas) [inline addons/sys_signal/XEH_preInit.sqf]: This should be a separate function, but maybe leave that for another PR.

File: addons/sys_sounds/XEH_postInit.sqf [hunk 13]
```diff
@@ -1,4 +1,5 @@
 #include "script_component.hpp"
-NO_DEDICATED;
 
-["handleLoadedSound", FUNC(handleLoadedSound)] call EFUNC(sys_rpc,addProcedure);
+if (hasInterface) then {
```
Comment (jonpas) [inline addons/sys_sounds/XEH_postInit.sqf]: Let's keep `!hasInterface -> exitWith` here as well.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_signal/XEH_preInit.sqf (hunk 12 above)
Comment (TheMagnetar) [reply addons/sys_signal/XEH_preInit.sqf]: This particular instance of handleEndTransmission is not used anywhere else. Maybe it should be removed.

File: addons/sys_attenuate/XEH_preInit.sqf (hunk 4 above)
Comment (jonpas) [reply addons/sys_attenuate/XEH_preInit.sqf:5]: Solve but is not on this line so this thing does not disappear!

**Review by jonpas — APPROVED** [review]

File: addons/sys_signal/XEH_preInit.sqf (hunk 12 above)
Comment (TheMagnetar) [reply addons/sys_signal/XEH_preInit.sqf]: Moved to another file. See #208

**Pull request merged**